auth/src/main/java/org/aerogear/mobile/security/SyncSecurityCheckExecutor.java

```diff
@@ -39,40 +41,44 @@ public SyncSecurityCheckExecutor build() {
      * Executes the provided checks and returns the results.
      * Blocks until all checks are executed.
      *
-     * @return the results of the executed checks
+     * Returns a {@link Map} containing the results of each executed test.
+     * The key of the map will be the output of {@link SecurityCheck#getName()}, while the value will be
+     * the result of the check.
+     *
+     * @return a {@link Map} containing the results of all the executed checks
      */
-    public SecurityCheckResult[] execute() {
-        SecurityCheckResult[] results = getTestResults();
+    public Map<String, SecurityCheckResult> execute() {
+        final Map<String, SecurityCheckResult> results = getTestResults();
         if (getMetricsService() != null) {
-            publishResultMetrics(results);
+            publishResultMetrics(results.values());
         }
         return getTestResults();
     }
 
     /**
      * Publish each {@link SecurityCheckResult result} provided as an {@link SecurityCheckResultMetric}.
      *
-     * @param results Array of results
+     * @param results Collection of results
      */
-    private void publishResultMetrics(final SecurityCheckResult[] results) {
+    private void publishResultMetrics(@NonNull final Collection <SecurityCheckResult> results) {
         for(SecurityCheckResult result : results) {
             this.getMetricsService().publish(new SecurityCheckResultMetric(result));
         }
     }
 
     /**
-     * Return the {@link SecurityCheckResult results} for each of the tests added.
+     * Returns a {@link Map} containing the results of each executed test.
+     * The key of the map will be the output of {@link SecurityCheck#getName()}, while the value will be
+     * the result of the check.
      *
-     * @return Array of results.
+     * @return a {@link Map} containing the results of all the executed checks
      */
-    private SecurityCheckResult[] getTestResults() {
+    private Map<String, SecurityCheckResult> getTestResults() {
 
-        Collection<SecurityCheck> checks = getChecks();
+        final Map<String, SecurityCheckResult> results = new HashMap<>();
 
-        SecurityCheckResult[] results = new SecurityCheckResult[checks.size()];
-        int i = 0;
-        for (SecurityCheck check : checks) {
-            results[i++] = check.test(getContext());
+        for (SecurityCheck check : getChecks()) {
+            results.put(check.getName(), check.test(getContext()));
```

Comment:
does a `SecurityCheck` have a name? or is the (simple)clazz name efficent enough ?

Reply:
@matzew The issue here is that the user must have an easy way to map the received result to the check that produced it. In my previous comment I suggest that the check name is the 'class name' to avoid name clash when the user implements a custom check.